src/backend/cdb/cdbpartition.c

```diff
@@ -5228,7 +5228,18 @@ get_part_rule(Relation rel,
 
 		lc = list_head(l1);
 		prule2 = (PgPartRule *) lfirst(lc);
-		if (prule2 && prule2->topRule && prule2->topRule->children)
+
+		if (!prule2)
+		{
+			if (!bExistError || !bMustExist)
+				return NULL;
+			else
+				ereport(ERROR,
```

Comment:
Can you provide a test case for these cases so that the user sees such an error?

Reply:
If `prule2 = (PgPartRule *) lfirst(lc)` results in `prule2` being `NULL`, this will be a bug, that will tell us that `AT_AP_IDRule` struct members were incorrectly set up. If `AT_AP_IDRule` is filled correctly, `prule2 == NULL` will never happen.

For `AT_AP_IDList`, I have replaced `ereport()` with `Insist(0)`, since we get rules from `get_part_rule1()`, and it can't return `NULL` if we are passing `bExistError = true`, since it will throw an error if rule does not exist.